gpt4all-chat/src/chatmodel.h

```diff
@@ -123,14 +146,29 @@ class ChatModel : public QAbstractListModel
         roles[ThumbsDownStateRole] = "thumbsDownState";
         roles[SourcesRole] = "sources";
         roles[ConsolidatedSourcesRole] = "consolidatedSources";
+        roles[PromptAttachmentsRole] = "promptAttachments";
         return roles;
     }
 
-    void appendPrompt(const QString &name, const QString &value)
+    void appendPrompt(const QString &name, const QString &value, const QList<QUrl> &attachedUrls)
     {
         ChatItem item;
         item.name = name;
         item.value = value;
+
+        for (const QUrl &url : attachedUrls) {
+            Q_ASSERT(url.isLocalFile());
+            const QString localFilePath = url.toLocalFile();
+            const QFileInfo info(localFilePath);
+            Q_ASSERT(info.suffix() == "xlsx"); // We only support excel right now
+            Q_ASSERT(info.exists());
+            Q_ASSERT(info.isFile());
```

Comment:
This is a TOCTOU issue. It's only necessary to check that the file exists when it's added and when we process the prompt. Assuming that the file still exists just to serialize the filename is an unnecessary reason for the program to crash.